code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/models/CarouselPushTemplate.kt

```diff
@@ -0,0 +1,158 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.services.ui.notification.models
+
+import android.content.Intent
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.services.ui.notification.PushTemplateConstants
+import com.adobe.marketing.mobile.util.DataReader
+import com.adobe.marketing.mobile.util.DataReaderException
+import com.adobe.marketing.mobile.util.JSONUtils
+import org.json.JSONArray
+import org.json.JSONException
+
+internal open class CarouselPushTemplate : AEPPushTemplate {
+    // Optional, Determines how the carousel will be operated. Valid values are "auto" or "manual".
+    // Default is "auto".
+    internal var carouselOperationMode: String
+        private set
+
+    // Required, One or more Items in the carousel defined by the CarouselItem class
+    internal var carouselItems = mutableListOf<CarouselItem>()
+        private set
+
+    // Required, "default" or "filmstrip"
+    internal var carouselLayoutType: String
+        private set
+
+    // Contains the carousel items as a string
+    internal var rawCarouselItems: String
+        private set
+
+    data class CarouselItem(
+        // Required, URI to an image to be shown for the carousel item
+        val imageUri: String,
+        // Optional, caption to show when the carousel item is visible
+        val captionText: String?,
+        // Optional, URI to handle when the item is touched by the user. If no uri is provided for the item, adb_uri will be handled instead.
+        val interactionUri: String?
+    )
+
+    constructor(data: Map<String, String>) : super(data) {
+        try {
```

Comment:
nit: I would prefer switching to `optString` and a check to see if the returned value is null or empty.